### PR TITLE
M1 support

### PR DESCRIPTION
**Pull request by Tom-Ski**
- GHA is building arm64 slices
- Library loader detecting arm and 64bit to change choice of native lib
- gdx-jnigen is updated to 2.2.1
- Freetype is now being compiled and built in GHA into a fat library, and then used to link against when building the bindings.

### CODE REVIEW

**Comment by SpaiR** [comment]
You need to configure an additional matrix build for arm. As it's done for all platforms. Like, no need to build a separate freetype for an amd build. As well as to configure an arm workaround when there is no need of it. Also, check job artefacts and you'll see that now there is only one dylib - the arm one. When it should be two of them.

And don't add to the PR compile arm binary, they are updated through the CI.

**Review by SpaiR — CHANGES_REQUESTED** [review]
You need to configure an additional matrix build for arm. As it's done for all platforms. Like, no need to build a separate freetype for an amd build. As well as to configure an arm workaround when there is no need of it. Also, check job artefacts and you'll see that now there is only one dylib - the arm one. When it should be two of them.

And don't add to the PR compiled arm binary, they are updated through the CI.

**Comment by Tom-Ski** [comment]
Do you want to separate this at the generateLibs level? Atm type from the matrix is passed directly in, but another entry for arm will need that part adapting too.

Or should we just do some logic and feed in 'mac' still to generateLibs for that arm type?

Im not sure what you mean by not needing a separate build for arm, that is required. Do you mean no need to separate it when adding a new entry to the build matrix?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -62,6 +66,29 @@ jobs:
...
+        name: Download FreeType - Compile & Install
+        working-directory: /opt/freetype
+        run: |
+          # freetype download and untar and install
+          sudo wget -O freetype.tar.gz ${{ env.FREETYPE_URL }}
+          sudo tar -xzf freetype.tar.gz -C . --strip-components=1
+          
+          sudo ./configure CFLAGS="-arch arm64 -arch x86_64" --with-zlib=no --with-bzip2=no --with-png=no --with-harfbuzz=no --with-brotli=no
+          sudo make
+          sudo make install
```
Comment (SpaiR) [inline .github/workflows/ci.yml:90]: You don't need it for amd64 builds.

**Comment by Displee** [comment]
Can this be merged?

**Comment by SpaiR** [comment]
Closing in favor of #223
I'll mention this PR in changelog as well. Thanks for the contrib! 🙏

**Pull request closed without merging**